### PR TITLE
Change i18n spec that detects unused keys to only fail if English has unused keys.

### LINKED ISSUE #1161: Should we stop using i18n-tasks for unused keys in non-English locales?
@jonatanklosko suggested that we remove the test for unused keys in our non-en locale files.
The motivation behind that is that we would stop having to edit existing locales (other than English), and that we would have unused keys only for a while, since updates to the locales through the app will remove them.

### PR DESCRIPTION
**Pull request by jfly**
This will make it easier for developers to do their jobs, rather than
having to troll through translations deleting lines (see #1201). Also updated /server-status to
show these unused keys in languages other than English. That way, we'll
know about these unused keys, but it does not force our developers to
delete them.
This fixes #1161.

Fun screenshots:

image

image

### CODE REVIEW

File: WcaOnRails/app/helpers/server_status_helper.rb [hunk 1]
```diff
@@ -1,28 +1,14 @@
...
-    end
-    [class_heading, class_panel, class_missing, class_outdated, title]
+  def css_classes(missing_keys_empty, unused_keys_empty, outdated_keys_empty)
+    class_missing = missing_keys_empty ? "success" : "danger"
+    class_unused = unused_keys_empty ? "success" : "warning"
+    class_outdated = outdated_keys_empty ? "success" : "warning"
+
+    all_good = missing_keys_empty && unused_keys_empty && outdated_keys_empty
+    class_panel = all_good ? "success" : "warning"
+    title = all_good ? "All good" : "Needs attention"
```
Comment (jfly) [inline WcaOnRails/app/helpers/server_status_helper.rb]: With a 3rd boolean added to this method, we end up with `2**3 = 8` possible states for the translation. I didn't feel like writing something clever to turn this into a sentence, so I just went with "Needs attention". @viroulep, you went to all the effort of writing the previous sentences, are you okay with this simplification?

File: WcaOnRails/spec/models/locale_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    @es = Locale.new('es', true)
+  end
+
+  it "#compare_to works" do
+    missing, unused, outdated = @es.compare_to(@en)
+    expect(missing).to match_array [ "layer1 > just_added", "cubes" ]
+    expect(unused).to match_array [ "layer1 > please_remove_me", "magics" ]
+    expect(outdated).to match_array [ "layer1 > recently_changed" ]
+  end
+end
```
Comment (jonatanklosko) [inline WcaOnRails/spec/models/locale_spec.rb]: These are mocks, so I was certainly wrong. Ignore that.

File: WcaOnRails/app/views/server_status/index.html.erb [hunk 3]
```diff
@@ -33,15 +33,16 @@
   </div>
   <% (I18n.available_locales - [:en]).each do |locale| %>
     <% missing = @status_locales[locale][:missing] %>
+    <% unused = @status_locales[locale][:unused] %>
     <% outdated = @status_locales[locale][:outdated] %>
-    <% all_empty = (missing.empty? && outdated.empty?) %>
-    <% class_heading, class_panel, class_missing, class_outdated, title = css_classes(missing.empty?, outdated.empty?) %>
+    <% all_empty = (missing.empty? && unused.empty? && outdated.empty?) %>
```
Comment (jfly) [inline WcaOnRails/app/views/server_status/index.html.erb]: Yes it is! I have an idea for cleaning this up, brb.

File: WcaOnRails/spec/models/locale_spec.rb (hunk 2 above)
Comment (jfly) [inline WcaOnRails/spec/models/locale_spec.rb]: Just discussed with Jonatan offline: there aren't actually any real files involved in this test, as I'm mocking out the call to `File.read` above.

**Comment by jfly** [comment]
Just refactored `@status_locales` into `@bad_keys_by_type_by_locale`. I'm very pleased with the cleanup. The only downside is that we now treat all types of bad keys as "dangerous", whereas previously, only missing keys were considered dangerous, unused and outdated were merely warnings.

In case the vocabularly I'm using is confusing, a "bad key" is a translation that needs attention. All bad keys are one of 3 types: "missing", "unused", or "outdated".

File: WcaOnRails/app/helpers/server_status_helper.rb (hunk 1 above)
Comment (viroulep) [inline WcaOnRails/app/helpers/server_status_helper.rb]: That's fine :)

File: WcaOnRails/app/views/server_status/index.html.erb [hunk 4]
```diff
@@ -32,67 +32,52 @@
     </div>
   </div>
   <% (I18n.available_locales - [:en]).each do |locale| %>
-    <% missing = @status_locales[locale][:missing] %>
-    <% outdated = @status_locales[locale][:outdated] %>
-    <% all_empty = (missing.empty? && outdated.empty?) %>
-    <% class_heading, class_panel, class_missing, class_outdated, title = css_classes(missing.empty?, outdated.empty?) %>
+    <%
+      bad_keys_by_type = @bad_keys_by_type_by_locale[locale]
+      bad_keys = bad_keys_by_type.values.flatten
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/server_status/index.html.erb]: We can use `all_bad_keys` to not override this below (in the block). I'm not strong about this, so do whatever seems appropriate for you.

**Comment by jonatanklosko** [comment]
LGTM =)

**Pull request merged**